client/workers/image-loader.worker.js

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-restricted-globals */
+
+self.addEventListener('message', (event) => {
+  const imageURL = event.data
+  let response = {}
+
+  imageURL.forEach(async (imageData) => {
+    response = await fetch(imageData[1])
+    const blob = await response.blob()
+
+    self.postMessage({
+      id: imageData[0],
+      payload: URL.createObjectURL(blob),
+    })
+  })
```

Comment:
Finally.. I think everything within the forEach should be wrapped in a try-catch just so it doesn't break if an image fails to load. Perhaps respond with a null payload if it fails.

Reply:
Should I respond with null or the error itself?

Reply:
I'd respond with 

```javascript
{
  id,
  payload: null,
}
```
 for now.

We could probably respond with an error in the future once we get some sort of error display alert thing going like we did at RollForGuild, but thats another day.